evm/src/main/java/org/hyperledger/besu/evm/processor/MessageCallProcessor.java

```diff
@@ -89,18 +90,25 @@ protected void codeSuccess(final MessageFrame frame, final OperationTracer opera
    * of the world state of this executor.
    */
   private void transferValue(final MessageFrame frame) {
-    final MutableAccount senderAccount =
-        frame.getWorldUpdater().getSenderAccount(frame).getMutable();
+    final EvmAccount senderAccount = frame.getWorldUpdater().getSenderAccount(frame);
```

Comment:
I think it might help to have some more comments? explaining what's going on - so we only need a mutable account if we're _actually_ transferring value ie value > 0

and if we're accessing public state from private state then value == 0 AND nothing is or needs to be mutable

Reply:
> I think it might help to have some more comments?

I'm always a bit reticent to put comments in. But in this case, you're right it's not clear what the intent/reason was.